app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -141,7 +141,13 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
             buildInfo = GleanBuildInfo.buildInfo
         )
 
-        setStartupMetrics(components.core.store, settings())
+        // We avoid blocking the main thread on startup by setting startup metrics on the background thread.
+        val isLoggedIn =
+            components.backgroundServices.accountManager.accountProfile() != null
```

Comment:
Moving the account manager to this place would also early initialize it during the `Application` creation which we would want to avoid.

I think using a pref to store the `loggedIn` state when the account manager is done initializing might be a quick fix for this. It's something we're doing for other settings that we want to submit to the ping as well. All future reads of this pref should give you the right value.

An alternative, is if observe the account manager and submit all these metrics when everything is done. I'm not sure how Glean SDK is required to work to know if this will cause faults in our data though, we're already trying to solve this issue right now after all.

Reply:
Good catch, I'll look into solving this issue.